components/global/hooks.tsx

```diff
@@ -99,3 +101,18 @@ export const useShowTutorialStep = (stepToShow: number, products?: ProductCompon
 
     return showTutorialStep;
 };
+
+/**
+ * Hook that returns the current open state of the specified modal
+ * - returns both the direct boolean for regular use and a ref that contains the boolean for usage in a callback
```

Comment:
For my own reference, why return a ref here too?

Reply:
So you can't use the directly returned value from the hook in the eventListener callback as it will only have access to the original value and not any of the subsequent updates. My research into it a while back indicated you need to use a ref in that case, but if you don't need to reference it from this kind of callback, it's better to use the direct value. Since that is the most likely use case of this kind of hook, it's returned first.

I initially thought that the re-rendering of the component would have provided the callback with the new value, but the component doesn't re-render on native DOM events so ...

I did try again anyway, but no dice.

If you have any ideas for a better way to handle this particular kind of situation, I'm all ears!